File: backend/src/main/java/com/roadmap/Roadmap/models/Quiz.java
package com.roadmap.Roadmap.models;

import java.util.Vector;

import org.springframework.data.annotation.Id;

public class Quiz {
    public class Question{
        public String question;
        public String answer;
    }
    @Id
    public String id;
    public String name;
    public Vector<Question> questions; 
}

File: backend/models/Quiz.js
const mongoose = require("mongoose");

const { Schema } = mongoose;

const QuizSchema = new Schema({
  name: String,
  questions: [
    {
      question: String,
      answer: String,
    },
  ],
});

const Quiz = mongoose.model("Quiz", QuizSchema);

module.exports = Quiz;
